test/Microsoft.AspNetCore.Razor.Test/Compilation/TagHelpers/TagHelperDescriptorTest.cs

```diff
@@ -427,6 +472,44 @@ public void TagHelperDescriptor_WithIndexerAttributes_CanBeDeserialized()
             Assert.Equal(expectedDescriptor.AssemblyName, descriptor.AssemblyName, StringComparer.Ordinal);
             Assert.Equal(expectedDescriptor.Attributes, descriptor.Attributes, TagHelperAttributeDescriptorComparer.Default);
             Assert.Empty(descriptor.RequiredAttributes);
+            Assert.Empty(descriptor.PropertyBag);
+        }
+
+        [Fact]
+        public void TagHelperDescriptor_WithPropertyBagElements_CanBeDeserialized()
+        {
+            // Arrange
+            var serializedDescriptor =
+                $"{{\"{nameof(TagHelperDescriptor.Prefix)}\":\"prefix:\"," +
+                $"\"{nameof(TagHelperDescriptor.TagName)}\":\"tag name\"," +
+                $"\"{nameof(TagHelperDescriptor.TypeName)}\":\"type name\"," +
+                $"\"{nameof(TagHelperDescriptor.AssemblyName)}\":\"assembly name\"," +
+                $"\"{ nameof(TagHelperDescriptor.PropertyBag) }\":" +
+                    "{\"key one\":\"value one\",\"key two\":\"value two\"}}";
+            var expectedDescriptor = new TagHelperDescriptor
+            {
+                Prefix = "prefix:",
+                TagName = "tag name",
+                TypeName = "type name",
+                AssemblyName = "assembly name"
+            };
+
+            expectedDescriptor.PropertyBag.Add("key one", "value one");
+            expectedDescriptor.PropertyBag.Add("key two", "value two");
+
+            // Act
+            var descriptor = JsonConvert.DeserializeObject<TagHelperDescriptor>(serializedDescriptor);
+
+            // Assert
+            Assert.NotNull(descriptor);
+            Assert.Equal(expectedDescriptor.Prefix, descriptor.Prefix, StringComparer.Ordinal);
+            Assert.Equal(expectedDescriptor.TagName, descriptor.TagName, StringComparer.Ordinal);
+            Assert.Equal(expectedDescriptor.TypeName, descriptor.TypeName, StringComparer.Ordinal);
+            Assert.Equal(expectedDescriptor.AssemblyName, descriptor.AssemblyName, StringComparer.Ordinal);
+            Assert.Empty(descriptor.Attributes);
+            Assert.Empty(descriptor.RequiredAttributes);
+            Assert.Equal(expectedDescriptor.PropertyBag.Keys, descriptor.PropertyBag.Keys);
```

Comment:
Dictionaries do not guarantee preserving ordering of keys and values. Might be safer to just check for the two keys you added. In this case, this would look like:

```C#
Assert.Equal(expectedDescriptor.PropertyBag["key one"], descriptor.PropertyBag["key one"]);
Assert.Equal(expectedDescriptor.PropertyBag["key two"], descriptor.PropertyBag["key two"]);
```